pages/company/[filename].tsx

```diff
@@ -0,0 +1,119 @@
+import { Layout } from "../../components/layout";
+import { Section } from "../../components/util/section";
+import { Breadcrumbs } from "../../components/blocks/breadcrumbs";
+import { removeExtension } from "../../services/client/utils.service";
+import { tinaField, useTina } from "tinacms/dist/react";
+import { BuiltOnAzure } from "../../components/blocks";
+import { RecaptchaContext } from "../../context/RecaptchaContext";
+import HistoryTimeline from "../../components/company/historyTimeline";
+import { HistoryTimelineCardProps } from "../../components/company/historyTimelineCard";
+import { TinaMarkdown } from "tinacms/dist/rich-text";
+import { componentRenderer } from "../../components/blocks/mdxComponentRenderer";
+
+export default function CompanyPage(
+  props: InferGetStaticPropsType<typeof getStaticProps>
+) {
+  const { data } = useTina({
+    data: props.data,
+    query: props.query,
+    variables: props.variables,
+  });
+
+  const historyCardProps =
+    data.company?.historyCards?.map<HistoryTimelineCardProps>((m) => ({
+      year: m.year,
+      title: m.title,
+      location: m.location as HistoryTimelineCardProps["location"],
+      description: m.description,
+    })) || [];
+
+  return (
+    <div>
+      <RecaptchaContext.Provider
+        value={{ recaptchaKey: props.env.GOOGLE_RECAPTCHA_SITE_KEY }}
+      >
+        <SEO seo={props.seo} />
+        <Layout>
+          <Section className="mx-auto w-full max-w-9xl px-8 py-5">
+            <Breadcrumbs
+              path={removeExtension(props.variables.relativePath)}
+              suffix={data.global.breadcrumbSuffix}
+              title={data.company.seo?.title}
+              seoSchema={data.company.seo}
+            />
+          </Section>
+          {data.company._body.children.length > 0 && (
+            <Section className="mx-auto w-full max-w-9xl px-8 py-5">
+              <div data-tina-field={tinaField(data.company, "_body")}>
+                <TinaMarkdown
+                  components={componentRenderer}
+                  content={data.company._body}
+                />
+              </div>
+            </Section>
+          )}
+          {data.company.historyCards?.length > 0 ? (
+            <Section className="mx-auto w-full max-w-9xl px-8 py-5">
+              <HistoryTimeline cardProps={historyCardProps} />
+            </Section>
+          ) : (
+            <></>
+          )}
+          <Section>
+            <BuiltOnAzure data={{ backgroundColor: "default" }} />
+          </Section>
+        </Layout>
+      </RecaptchaContext.Provider>
+    </div>
+  );
+}
+
+export const getStaticProps = async ({ params }) => {
+  const tinaProps = await client.queries.companyContentQuery({
+    relativePath: `${params.filename}.mdx`,
+  });
+
+  const canonical = `${tinaProps.data.global.header.url}company/${params.filename}`;
+  const seo = tinaProps.data.company.seo;
+  if (seo) {
+    seo.canonical = canonical;
+  }
```

Comment:
Understand what this does, do we need it?

Reply:
@bradystroud 

From [this article](https://moz.com/learn/seo/canonicalization), a canonical url prevents crawlers indexing a site multiple times when the url is slightly different, ie http:// => https:// or /consulting/cms => /consulting/cms?r... 

This piece of code is setting that canonical url on each page, which we can see in the head tag of each page:

![image](https://github.com/SSWConsulting/SSW.Website/assets/10693364/a3b6cec2-4295-40f5-bb5d-1dd63be1d8ff)
**Figure: The canonical url element in the head tag of `/consulting/cms`**

We should probably make a task to check each template to ensure they have one. It looks like some such as the [training template](https://www.ssw.com.au/events/angular-superpowers-tour) are missing it.